### PR TITLE
New feature in Git 3: closures

### PR DESCRIPTION
**Pull request by KvanTTT**

### CODE REVIEW

File: Git-Closure/English.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
+# New feature in Git 3: closures
+
+Git is a popular version control system. In Git, an atomic change of one or
+several files is called a commit, and several consecutive commits are combined
+into a branch. Branches are used to implement new ideas (features).
+
+![Git Logo](Logo.svg)
+
+It happens that the idea is a dead end and the developer has turned the wrong
```
Comment (gabriel-fallen) [inline Git-Closure/English.md:9]: "an idea", you're not talking about single specific one.

File: Git-Closure/English.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+way, so he needs to roll back to the original version. He should forget about
+the new branch, switch to the main **dev** or **master** branch and continue
+working. In this case, the "scion" will hang forever, as well as the desire to
+remove it. But how to remove the part of the history? This branch shows the
```
Comment (gabriel-fallen) [inline Git-Closure/English.md:13]: "a part of the history".

File: Git-Closure/English.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+How to use this command, what benefits does it give and what do IT companies
+think? The article answers these and other questions.
+
+<cut/>
+
+## Description
+
+Now it is possible to close an unsuccessful branch over one of the previous
+commits. Closure arcs are yellow colored in the pictures below.
```
Comment (gabriel-fallen) [inline Git-Closure/English.md:29]: I'd suggest "Closure arcs are painted in yellow in the pictures below." or something along these lines.

File: Git-Closure/English.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+-->
+
+## How to use?
+
+The `merge` command does not include the functionality of closures, since for
+the first case the branch will be
+[fast-forwarded](https://stackoverflow.com/q/9069061/1046374), and for the
+second case nothing will be done (`git already up to date`).
+
+In order not to change the old behavior, the developers decided to introduce a
```
Comment (gabriel-fallen) [inline Git-Closure/English.md:94]: I think the comma isn't needed here.

File: Git-Closure/English.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+![Git Multiple Closure](Git-Closure-Multiple.svg)
+
+After the command is executed, Git calculates the changes, and in the final
+commit a double diff will be shown: from the base and closing branches. In the
+general case, it is an n-dimensional diff, that is, there can be as many
+closures as you wish. closure-commit is similar to merge-commit with the only
+difference that it contains several messages, not one.
+
+Unfortunately, existing Git GUI's do not have good support of closures.
```
Comment (gabriel-fallen) [inline Git-Closure/English.md:118]: "Git GUIs", apostrophe is misleading here.
"good support for closures".

File: Git-Closure/English.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+[GitExtensions](https://github.com/gitextensions/gitextensions) preview version
+displays merge curves instead of elegant arcs. Take a look on the new fields such
```
Comment (gabriel-fallen) [inline Git-Closure/English.md:120]: "Take a look at".

File: Git-Closure/English.md [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+recalculating commits is complicated.
+
+Also the `auto` option allows automatic closures of all the old branches. In
+this case, the closing commit is the one starting the branch. With Git IDE
+plugins, closures can be run periodically. In
+GitExtensions there is a similar plugin **Delete obsolete branches**.
+
+## What IT companies think
+
+Large IT companies: Google, Facebook, Apple, DeepMind, Positive Technologies,
```
Comment (gabriel-fallen) [inline Git-Closure/English.md:139]: I like Positive Technologies in this list, though I strongly suspect it's dwarfed by these peers in terms of market capitalization. :smile:

File: Git-Closure/English.md [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+problem is not only ours. Closures not only improve organization, but also make
+it possible to track programmer's reasoning, sometimes completely
+incomprehensible even to colleagues :) Closures reminded me of the movie "Back
+to the Future", where the characters traveled to the past and the future. I like
+this movie, I watched it several times. And I think I will love Git even more
+because if this feature :)
+
+## Note
+
+If earlier the graph of commits was a [directed acyclic
```
Comment (gabriel-fallen) [inline Git-Closure/English.md:159]: "If previously" I think is a better term...

**Review by gabriel-fallen — COMMENTED** [review]
You employ way too many commas in English text and place them according to the rules of Russian language! :smiley:

File: Git-Closure/English.md (hunk 7 above)
Comment (KvanTTT) [reply Git-Closure/English.md:139]: Sure, but I think it's ok for April 1 Fool's Day article :smile:

**Comment by KvanTTT** [comment]
Thanks for the review!

**Pull request merged**